File: packages/sdk-android/src/main/java/app/eodin/deeplink/EodinException.kt
package app.eodin.deeplink

/**
 * Exception types for Eodin SDK operations.
 */
sealed class EodinException(message: String) : Exception(message) {

    /**
     * SDK is not configured. Call EodinDeeplink.configure() first.
     */
    class NotConfigured : EodinException(
        "EodinDeeplink SDK is not configured. Call EodinDeeplink.configure() first."
    )

    /**
     * Network request failed.
     */
    class NetworkError(cause: Throwable) : EodinException(
        "Network error: ${cause.message}"
    ) {
        init {
            initCause(cause)
        }
    }

    /**
     * Invalid response from server.
     */
    class InvalidResponse : EodinException(
        "Invalid response from server"
    )

    /**
     * No deferred parameters found for this device.
     * This is expected for fresh installs without clicking a deep link.
     */
    class NoParamsFound : EodinException(
        "No deferred parameters found for this device"
    )

    /**
     * Server returned an error.
     *
     * @property code HTTP status code
     * @property serverMessage Error message from server
     */
    class ServerError(val code: Int, val serverMessage: String?) : EodinException(
        "Server error ($code): ${serverMessage ?: "Unknown error"}"
    )
}

File: packages/sdk-flutter/lib/src/exceptions/eodin_exception.dart
/// Base exception for Eodin SDK errors
class EodinException implements Exception {
  /// Creates a new [EodinException]
  const EodinException(this.message);

  /// Error message
  final String message;

  @override
  String toString() => 'EodinException: $message';
}

/// Exception thrown when SDK is not configured
class NotConfiguredException extends EodinException {
  /// Creates a new [NotConfiguredException]
  const NotConfiguredException()
      : super('EodinDeeplink is not configured. Call configure() first.');
}

/// Exception thrown when no deferred parameters are found
class NoParamsFoundException extends EodinException {
  /// Creates a new [NoParamsFoundException]
  const NoParamsFoundException()
      : super('No deferred parameters found for this device.');
}

/// Exception thrown when a network error occurs
class NetworkException extends EodinException {
  /// Creates a new [NetworkException]
  const NetworkException(this.error) : super('Network error occurred');

  /// The underlying error
  final dynamic error;

  @override
  String toString() => 'NetworkException: $message ($error)';
}

/// Exception thrown when the API returns an error
class ApiException extends EodinException {
  /// Creates a new [ApiException]
  const ApiException(this.statusCode, String message) : super(message);

  /// HTTP status code
  final int statusCode;

  @override
  String toString() => 'ApiException [$statusCode]: $message';
}
